index.js

```diff
@@ -315,14 +317,16 @@ function assertValidSchema(db, { docTableName, backlinkTableName }) {
 function assertMatchingSchema(tableName, columns, schema) {
   for (const [name, info] of Object.entries(schema)) {
     const column = columns.find((c) => c.name === name)
-    assert(column, `Table '${tableName}' must have a column '${name}'`)
-    for (const [prop, value] of Object.entries(info)) {
-      assert(
-        // @ts-ignore
-        column[prop] === value,
-        // @ts-ignore
-        `Column '${name}' in table '${tableName}' should have ${prop}=${value}, but instead ${prop}=${column[prop]}`
-      )
+    if (info.notnull) {
+      assert(column, `Table '${tableName}' must have a column '${name}'`)
+      for (const [prop, value] of Object.entries(info)) {
+        assert(
+          // @ts-ignore
+          column[prop] === value,
+          // @ts-ignore
+          `Column '${name}' in table '${tableName}' should have ${prop}=${value}, but instead ${prop}=${column[prop]}`
+        )
+      }
```

Comment:
Not sure why you are doing this? We should check that all the columns exist in the table.

Reply:
oh, I thought that since `deleted` was optional, then it shouldn't be created directly with the table